src/actors/objects/mod.rs

```diff
@@ -75,13 +71,24 @@ impl From<io::Error> for ObjectError {
     }
 }
 
+impl From<crate::services::download::DownloadError> for ObjectError {
```

Comment:
maybe `use crate::services::download::{DownloadError, DownloadErrorKind}`?

Reply:
same logic as before: this is used in one place, not core to this module.